app/controllers/application_controller.rb

```diff
@@ -17,7 +17,11 @@ class ApplicationController < ActionController::API
   end
 
   def paginate(scope)
-    scope.paginate(:per_page => params[:per_page], :page => params[:page])
+    scope.paginate(:per_page => per_page(params), :page => page(params))
+  end
+
+  def pagination_options(scope)
+    { :meta => { :total => scope.count, :per_page => per_page(params), :page => page(params) } }
```

Comment:
You can get `per_page` and `page` params out of the scope:
``` ruby
zz = App.paginate(per_page: 3, page: 2)
zz.current_page
# => 2
zz.per_page
# => 3

```

Reply:
Something new every day, thanks. Will change